src/widgets/geo-search/GeoSearchRenderer.js

```diff
@@ -196,14 +232,26 @@ const renderer = (
 
   // Fit the map to the markers when needed
   const hasMarkers = renderState.markers.length;
-  const center = renderState.mapInstance.getCenter();
-  const zoom = renderState.mapInstance.getZoom();
-  const isPositionInitialize = center !== undefined && zoom !== undefined;
-  const enableFitBounds =
-    !hasMapMoveSinceLastRefine() &&
-    (!isRefinedWithMap() || (isRefinedWithMap() && !isPositionInitialize));
-
-  if (hasMarkers && enableFitBounds) {
+  const boundingBoxFromConnector = getSearchBounds();
+  const enableFitBounds = !hasMapMoveSinceLastRefine() && !isRefinedWithMap();
+
+  if (boundingBoxFromConnector && isRefinedWithMap()) {
+    renderState.isUserInteraction = false;
+    const boundsWithPadding = getBoundsWithPadding({
+      bounds: boundingBoxFromConnector,
+      paddingBoundingBox,
+      mapInstance: renderState.mapInstance,
+      googleReference,
+    });
+    renderState.mapInstance.fitBounds(
+      new googleReference.maps.LatLngBounds(
+        boundsWithPadding.southWest,
+        boundsWithPadding.northEast
+      ),
+      0
+    );
+    renderState.isUserInteraction = true;
+  } else if (hasMarkers && enableFitBounds) {
```

Comment:
We can add a test for this case.

Reply:
Ok I will have a look.

Reply:
Can you tell me what kind of test do you have in mind here @samouss ?

Reply:
We can check that both cases correctly call fitBounds with the proper values. In the previous implementation the two tests removed does this kind of assertions.

Reply:
I added a test @samouss  :) Not the greatest, but it's okayish